src/app/(nextjs_migration)/(guest)/layout.tsx

```diff
@@ -2,23 +2,65 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+import { v5 as uuidv5 } from "uuid";
 import { ReactNode } from "react";
 import "../../../client/css/index.css";
 import Image from "next/image";
 import MonitorLogo from "../../../client/images/monitor-logo-transparent@2x.webp";
 import MozillaLogo from "../../../client/images/moz-logo-1color-white-rgb-01.svg";
 import { SignInButton } from "../components/client/SignInButton";
 import { getL10n } from "../../functions/server/l10n";
+import { getServerSession } from "next-auth";
+import { authOptions } from "../../api/utils/auth";
+import { isFlagEnabled } from "../../functions/server/featureFlags";
+import { cookies } from "next/headers";
+import { randomUUID } from "crypto";
+import { PageLoadEvent } from "../components/client/PageLoadEvent";
 
 export type Props = {
   children: ReactNode;
 };
 
-const GuestLayout = (props: Props) => {
+const GuestLayout = async (props: Props) => {
   const l10n = getL10n();
 
+  // If the user is logged in, use UUID derived from FxA UID as Nimbus user ID.
+  const session = await getServerSession(authOptions);
+  const accountId = session?.user?.subscriber?.fxa_uid;
+  let userId;
+
+  if (accountId) {
+    // If the user is logged in, use a UUID based on the user's subscriber ID.
+    // Note: we may want to use the FxA UID here, but we need approval for that first.
+    if (!process.env.NIMBUS_UUID_NAMESPACE) {
+      throw new Error("env var NIMBUS_UUID_NAMESPACE not set");
+    }
+    userId = uuidv5(accountId.toString(), process.env.NIMBUS_UUID_NAMESPACE);
+  } else {
+    // if the user is not logged in, use a cookie with a randomly-generated Nimbus user ID.
+    const cookie = cookies().get("userId");
+    if (cookie) {
+      userId = cookie.value;
+    } else {
+      // TODO Cookies can only be set in server action or route handler
+      // @see https://nextjs.org/docs/app/api-reference/functions/cookies#cookiessetname-value-options
+      // cookies().set("userId", uuid);
+      userId = `guest-${randomUUID()}`;
+    }
+  }
+
+  // @see https://github.com/mozilla/experimenter/tree/main/cirrus
+  const serverUrl = process.env.SERVER_URL ?? "http://localhost:6060";
+
+  //@ts-ignore TODO this tells us which features to enable, for A/A testing we do nothing.
```

Comment:
Same as other comment.

Reply:
I copy/pasted this but I will try to make this shared, same answer as above :)

Reply:
I'll also file a Jira ticket and link it which will provide more complete context. The real TODO action here is:

- pass targeting info (locale, geo, etc) to Nimbus
- process `feature` response from Nimbus, and enable any features that the user is eligible for

Nimbus knows which experiments are active and will do the sampling and determine eligibility based on the targeting criteria, Monitor just needs to enable the proper feature for the current user (conceptually similar to the way we do feature-flag allow-lists now, but I don't think we need to use the database for this - we can ask Nimbus as-needed).

Reply:
I'll also moved the shared code to a server function, I didn't realize when I started that I'd need to implement this separately for guest/authenticated.